### ISSUE TITLE
Dev repo does not get created for boot + gitops

### ISSUE THREAD

**Issue by hferentschik** [jenkins-x/jx#5336]
When installing Jenkins-X with gitops enabled using boot, the dev repo does not get created.

```
$ jx boot
...
STEP: verify-env command: /bin/sh -c jx step verify env in dir: .

validating git repository for dev at URL https://github.com/hf-bee-bot/environment-hardy-jx-private-env-dev.git
creating environment dev git repository for URL: https://github.com/hf-bee-bot/environment-hardy-jx-private-env-dev.git to namespace jx
validating git repository for production at URL https://github.com/hf-bee-bot/environment-hardy-jx-private-env-production.git
creating environment production git repository for URL: https://github.com/hf-bee-bot/environment-hardy-jx-private-env-production.git to namespace jx-production
Using Git provider github at https://github.com
? Using Git user name: hf-bee-bot
? Using organisation: hf-bee-bot
Creating repository hf-bee-bot/environment-hardy-jx-private-env-production
Creating Git repository hf-bee-bot/environment-hardy-jx-private-env-production
Pushed Git repository to https://github.com/hf-bee-bot/environment-hardy-jx-private-env-production

validating git repository for staging at URL https://github.com/hf-bee-bot/environment-hardy-jx-private-env-staging.git
creating environment staging git repository for URL: https://github.com/hf-bee-bot/environment-hardy-jx-private-env-staging.git to namespace jx-staging
Using Git provider github at https://github.com
? Using Git user name: hf-bee-bot
? Using organisation: hf-bee-bot
Creating repository hf-bee-bot/environment-hardy-jx-private-env-staging
Creating Git repository hf-bee-bot/environment-hardy-jx-private-env-staging
Pushed Git repository to https://github.com/hf-bee-bot/environment-hardy-jx-private-env-staging
```

The log says "creating environment dev git repository", but then the repo is not actually created.

**Comment by hferentschik** [comment]
The issue lies in `createEnvGitRepository` of _step_verify_environment.go_:

```go
if name == kube.LabelValueDevEnvironment || environment.Spec.Kind == v1.EnvironmentKindTypeDevelopment {
		if !o.InCDPipeline() && !o.InCluster() {
			err := o.prDevEnvironment(gitInfo.Name, gitInfo.Organisation, !publicRepo, userAuth, requirements, server)
			if err != nil {
				return errors.Wrapf(err, "creating dev environment for %s", gitInfo.Name)
			}
		}
	}
```

`!o.InCDPipeline()` will always be false, no matter whether we are executing locally or in cluster. The reason is a questionable `InCDPipeline()` implementation. For our purposes it should be enough to just test for  `o.InCluster()`.

**Comment by hferentschik** [comment]
fixed by PR #5285
